Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIWebNavigation.mm

```diff
@@ -421,6 +421,56 @@
     [manager run];
 }
 
+TEST(WKWebExtensionAPIWebNavigation, ErrorOccurred)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/"_s, { { { "Content-Type"_s, "text/html"_s } }, "<iframe src='/frame.html'></iframe>"_s } },
+        { "/frame.html"_s, { HTTPResponse::TerminateConnection::Yes } },
+    }, TestWebKitAPI::HTTPServer::Protocol::Http);
+
+    auto *backgroundScript = Util::constructScript(@[
+        // Setup
+        @"function errorListener(details) {",
+        // A subframe should have been the one to have the error.
+        @"  browser.test.assertFalse(details.frameId == 0)",
+        @"  browser.test.assertEq(details.parentFrameId, 0)",
+
+        // Get more information about the frame to verify the errorOccurred bit was set.
+        @"  browser.webNavigation.getFrame({ tabId: details.tabId, frameId: details.frameId }, function(frame) {",
+        @"    browser.test.assertEq(frame.parentFrameId, 0)",
+        @"    browser.test.assertTrue(frame.errorOccurred)",
+        // One thing to note here is that if the provisional load fails, there won't be a URL in the details.
```

Comment:
You can assert this:

```
browser.test.assertEq(typeof frame.url, 'undefined')
```

Or should it be empty string? Either way you should test for it.

Reply:
I believe it would be the empty string. I can add a test.